Source/WTF/wtf/glib/GSocketMonitor.cpp

```diff
@@ -47,8 +47,8 @@ gboolean GSocketMonitor::socketSourceCallback(GSocket*, GIOCondition condition,
     monitor->m_isExecutingCallback = false;
 
     if (monitor->m_shouldDestroyCallback) {
-        monitor->m_callback = nullptr;
         monitor->m_shouldDestroyCallback = false;
+        monitor->m_callback = nullptr;
```

Comment:
I think your solution is the best we can do without making GSocketMonitor refcounted. But please add a one liner comment to make sure we don't reintroduce this bug!